src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -302,20 +304,41 @@ Copyright (c) .NET Foundation. All rights reserved.
     ============================================================
                                         _GetAppHostPaths
 
-    Gets the path to the restored apphost, and if the restored apphost
-    was present, computes the path for the destination apphost.
+    Gets the path to apphost (restored via packages or in an apphost pack),
+    and computes the path for the destination apphost.
     ============================================================
      -->
   <Target Name="_GetAppHostPaths"
-          DependsOnTargets="ResolvePackageAssets"
+          DependsOnTargets="_GetAppHostPathsFromAssetFile;_GetAppHostPathFromAppHostPack"
           Condition="'$(UseAppHost)' == 'true' and '$(_IsExecutable)' == 'true'">
+    
+  </Target>
+
+  <Target Name="_GetAppHostPathFromAppHostPack"
+          Condition="'$(UseAppHostFromAssetsFile)' == 'false'">
+
+    <PropertyGroup>
+      <!-- AppHostSourcePath already set by ResolveFrameworkReferences -->
+      <AppHostIntermediatePath>$(IntermediateOutputPath)$(AssemblyName)$([System.IO.Path]::GetExtension($(AppHostSourcePath)))</AppHostIntermediatePath>
```

Comment:
If we do keep this as three targets, then this needs to be an absolute path to workaround an issue we were seeing with the Resource API used in the `ResourceUpdater` (I can't seem to find the issue, but it was reported from a user).  This is why we previously used `@(_NativeAppHostNETCore->'%(FullPath)')`.

Reply:
Found the issue: https://github.com/dotnet/sdk/issues/2599